src/azul/plugins/repository/tdr/__init__.py

```diff
@@ -70,3 +111,192 @@ def dss_deletion_subscription_query(self, prefix: str) -> JSON:
+                entity_ids.remove(row[entity_type + '_id'])
+            if entity_ids:
+                raise RuntimeError(f'Required entities not found in {self._source.bq_name}.{entity_type}: {entity_ids}')
+        return bundle
+
+
+class Checksums(NamedTuple):
+    crc32c: str
+    sha1: str
+    sha256: str
+    s3_etag: str
+
+    def asdict(self) -> Dict[str, str]:
+        return self._asdict()
+
+    @classmethod
+    def without_values(cls) -> Dict[str, str]:
+        return {f: None for f in cls._fields}
+
+    @classmethod
+    def extract(cls, json: JSON) -> 'Checksums':
+        return cls(**{f: json[f] for f in cls._fields})
+
+
+@attr.s(auto_attribs=True, kw_only=True)
+class TDRBundle(Bundle):
+
+    def add_entity(self, entity_key: str, entity_type: str, entity_row: BigQueryRow) -> None:
+        content_type = 'links' if entity_type == 'links' else entity_row['content_type']
+        self.manifest.append(self._manifest_entry(name=entity_key,
+                                                  uuid=entity_row[entity_type + '_id'],
+                                                  version=entity_row['version'].strftime(Plugin.timestamp_format),
+                                                  size=entity_row['content_size'],
+                                                  content_type='application/json',
+                                                  dcp_type=f'"metadata/{content_type}"'))
+        if entity_type.endswith('_file'):
+            descriptor = json.loads(entity_row['descriptor'])
+            drs_path = None if entity_row['drs_template'] is None else entity_row['drs_template'].rsplit('/', 1)[1]
+            self.manifest.append(self._manifest_entry(name=entity_row['file_name'],
+                                                      uuid=descriptor['file_id'],
+                                                      version=descriptor['file_version'],
+                                                      size=descriptor['size'],
+                                                      content_type=descriptor['content_type'],
+                                                      dcp_type='data',
+                                                      checksums=Checksums.extract(descriptor),
+                                                      drs_path=drs_path))
+        self.metadata_files[entity_key] = json.loads(entity_row['content'])
+
+    @property
+    def metadata_columns(self) -> Set[str]:
+        return {
+            'version',
+            'JSON_EXTRACT_SCALAR(content, "$.schema_type") AS content_type',
+            'BYTE_LENGTH(content) AS content_size',
+            'content'
+        }
+
+    @property
+    def data_columns(self) -> Set[str]:
+        return self.metadata_columns | {
+            'descriptor',
+            'JSON_EXTRACT_SCALAR(content, "$.file_core.file_name") AS file_name',
+            'file_id AS drs_template'  # column is present but null for datasets
```

Comment:
I find drs_template oddly generic and misleading. If `file_id` is taken, I would use tdr_file_id or drs_file_id.